src/sinks/appsignal/request_builder.rs

```diff
@@ -0,0 +1,83 @@
+use std::sync::Arc;
+
+use bytes::Bytes;
+use prost::Message;
+use vector_common::request_metadata::RequestMetadata;
+
+use super::{
+    config::AppsignalCredentials, encoding::AppsignalEncoder, service::AppsignalApiRequest,
+    sink::EventCollection, AppsignalSinkError,
+};
+use crate::{
+    event::EventFinalizers,
+    proto::vector as proto_vector,
+    sinks::util::{
+        metadata::RequestMetadataBuilder, request_builder::EncodeResult, Compression,
+        RequestBuilder,
+    },
+};
+
+pub struct AppsignalRequestBuilder {
+    encoder: AppsignalEncoder,
+    compression: Compression,
+    credentials: Arc<AppsignalCredentials>,
+}
+
+impl AppsignalRequestBuilder {
+    pub fn new(compression: Compression, credentials: Arc<AppsignalCredentials>) -> Self {
+        Self {
+            encoder: AppsignalEncoder,
+            compression,
+            credentials,
+        }
+    }
+}
+
+impl RequestBuilder<EventCollection> for AppsignalRequestBuilder {
+    type Metadata = EventFinalizers;
+    type Events = proto_vector::PushEventsRequest;
+    type Encoder = AppsignalEncoder;
+    type Payload = Bytes;
+    type Request = AppsignalApiRequest;
+    type Error = AppsignalSinkError;
+
+    fn compression(&self) -> Compression {
+        self.compression
+    }
+
+    fn encoder(&self) -> &Self::Encoder {
+        &self.encoder
+    }
+
+    fn split_input(
+        &self,
+        event_collection: EventCollection,
+    ) -> (Self::Metadata, RequestMetadataBuilder, Self::Events) {
+        let event_count = event_collection.events.len();
+        let encoded_events = proto_vector::PushEventsRequest {
+            events: event_collection.events,
+        };
+        let encoded_len = encoded_events.encoded_len();
+        let builder = RequestMetadataBuilder::new(
+            event_count,
+            encoded_len,
+            encoded_len, // this is fine as no JSON is being sent
```

Comment:
Because we're not sending JSON, I left this the same as the protobuf length. The comment in the vector sink says something similar: https://github.com/vectordotdev/vector/blob/a18bc5f8989ef35ad037d2d55f955f625f7cae71/src/sinks/vector/sink.rs#L65